lib/orocos/process.rb

```diff
@@ -1194,22 +1194,23 @@ def self.resolve_running_tasks(process, &block)
             else
                 # Get any task name from that specific deployment, and check we
                 # can access it either through the ior pipe
-                begin
-                    while (incoming_message = @ior_read_fd.read_nonblock(4096))
-                        @ior_message += incoming_message
```

Comment:
why do you need `ior_message` as a class member?

Reply:
In case WaitReadable is raised, I can resume reading the message when this method is called again